api/python/t4/api.py

```diff
@@ -208,23 +208,188 @@ def list_packages(registry=None):
     Returns:
         A list of strings containing the names of the packages
     """
-    registry = get_package_registry(fix_url(registry) if registry else None) + '/named_packages'
+    class PackageList:
+        """Display wrapper for list_packages"""
+
+        def __init__(self, pkg_info):
+            self.pkg_names = [info[0].replace(':latest', '') for info in pkg_info]
+            self._repr = self.create_str(pkg_info)
+
+        def __repr__(self):
+            return self._repr
+
+        def __iter__(self):
+            return iter(self.pkg_names)
 
-    registry_url = urlparse(registry)
+        def __len__(self):
+            return len(self.pkg_names)
+
+        def __contains__(self, item):
+            return item in self.pkg_names
+
+        @staticmethod
+        def _fmt_str(string, strlen):
+            """Formats strings to a certain width"""
+            if len(string) > strlen - 3:
+                return string[:strlen - 6] + '...' + '   '
+            else:
+                return string.ljust(strlen)[:strlen - 3] + '   '
+
+        @staticmethod
+        def _humanize_bytesize(nbytes):
+            """Turns raw byte count into a human readable bytesize"""
+            suffixes = ['B', 'KB', 'MB', 'GB', 'TB', 'PB']
+            i = 0
+            while nbytes >= 1024 and i < len(suffixes) - 1:
+                nbytes /= 1024
+                i += 1
+            filesize = (f'{nbytes:.2f}').rstrip('0').rstrip('.')
+            suffix = suffixes[i]
+            return f'{filesize} {suffix}'
+
+        def create_str(self, pkg_info):
+            """Generates a human-readable string representation of a registry"""
+            if pkg_info:
+                pkg_name_display_width = max(max([len(info[0]) for info in pkg_info]), 30)
+            else:
+                pkg_name_display_width = 30
+
+            out = (f"{self._fmt_str('PACKAGE', pkg_name_display_width)}"
+                   f"TOPHASH        "
```

Comment:
uses tabs here to separate columns. A user can override that in there terminal settings (2 or 4) etc